Core.Collectors/Collector/CollectorBase.cs

```diff
@@ -79,7 +80,12 @@ public async Task<bool> ProcessAsync(T collectionNode, long maxPageCount)
                 counter++;
                 HttpResponseMessage response = await batchingHttpRequest.NextResponseAsync(this.authentication).ConfigureAwait(false);
 
-                if (response.IsSuccessStatusCode)
+                if (response.StatusCode == HttpStatusCode.NoContent)
+                {
+                    // Responses with empty bodies cannot be deserialized
+                    continue;
```

Comment:
You know the semantics of NoContent better than I do. Should this be a "break" instead of "continue"? In other words, is it possible that there is pagination and some pages give "NoContent" response but there is still pages / batching available?

Reply:
changing to "break" makes sense to me:

- It's really a decision for each exposed API to make; but reason tells me there cannot be more pages of content after a No Content response.
- The hash endpoint is the first (and probably only) one that uses 204, and both options (continue / break) will support it